packages/webview_flutter/lib/webview_flutter.dart

```diff
@@ -469,6 +469,32 @@ class WebViewController {
   }
 }
 
+/// Manages cookies pertaining to all [WebView]s.
+class CookieManager {
+  factory CookieManager() {
+    if (_instance == null) {
+      final MethodChannel methodChannel =
```

Comment:
Since we can set mockMethodCallHandlers for testing, what is the value of providing the channel this way?

Reply:
My original thought was to make this a const field on the class. I wanted to be consistent with Battery plugin, i should maybe look through the test code there.

Reply:
Consistency is good to a point, but correctness is better

Reply:
curious, does this approach expose a way of injecting the wrong method channel?

Reply:
It's not necessary at all

Reply:
i agree that passing in a method channel is not needed for testing, but passing it though this way -- creating it on initialization vs having it static, should be equivalent?

Reply:
Generally making const fields is frowned upon. since the const is already canonicalized we are only adding overhead to the class. Furthermore, removing the initialization from the factory lets us cut down the code a bit with `??=`:

```
factory Foo() {
  return _instance ??= Foo._();
}
Foo._();
static Foo _instance;
```